src/Domain/DomainCreators/Disk.cpp

```diff
@@ -0,0 +1,106 @@
+    typename InitialRefinementSquare::type initial_refinement_square,
+    typename InitialGridPointsWedge::type initial_number_of_grid_points_wedge,
+    typename InitialGridPointsSquare::type initial_number_of_grid_points_square,
+    const OptionContext& /*context*/) noexcept
+    // clang-tidy: trivially copyable
+    : inner_radius_(std::move(inner_radius)),                 // NOLINT
+      outer_radius_(std::move(outer_radius)),                 // NOLINT
+      initial_refinement_wedge_(                              // NOLINT
+          std::move(initial_refinement_wedge)),               // NOLINT
+      initial_refinement_square_(                             // NOLINT
+          std::move(initial_refinement_square)),              // NOLINT
+      initial_number_of_grid_points_wedge_(                   // NOLINT
+          std::move(initial_number_of_grid_points_wedge)),    // NOLINT
+      initial_number_of_grid_points_square_(                  // NOLINT
+          std::move(initial_number_of_grid_points_square)) {  // NOLINT
+
+  ASSERT(initial_number_of_grid_points_square_[0] ==
+             initial_number_of_grid_points_square_[1],
+         "It is not yet possible for the square to not be a square - the "
+         "number of gridpoints must by the same in x and y. ");
+  ASSERT(initial_number_of_grid_points_square_[0] ==
+             initial_number_of_grid_points_wedge_[1],
+         "The number of angular grid points along each wedge must be the same "
+         "as the number of grid points on each side of the square.");
+  ASSERT(initial_refinement_square_[0] == initial_refinement_square_[1],
+         "It is not yet possible for the square to not be a square - the "
+         "refinement must by the same in x and y. ");
+  ASSERT(initial_refinement_square_[0] == initial_refinement_wedge_[1],
+         "The angular refinement along each wedge must be the same as the "
+         "refinement on each side of the square.");
+}
+Domain<2, Frame::Inertial> Disk::create_domain() const noexcept {
+  using Wedge2DMap = CoordinateMaps::Wedge2D;
+  using Wedge2DType =
+      CoordinateMap<Frame::Logical, Frame::Inertial, Wedge2DMap>;
+  using AffineMap = CoordinateMaps::AffineMap;
+  using AffineMap2D = CoordinateMaps::ProductOf2Maps<AffineMap, AffineMap>;
+
+  std::array<size_t, 4> block0_corners{{1, 5, 3, 7}},  //+x wedge
+      block1_corners{{3, 7, 2, 6}},                    //+y wedge
+      block2_corners{{2, 6, 0, 4}},                    //-x wedge
+      block3_corners{{0, 4, 1, 5}},                    //-y wedge
+      block4_corners{{0, 1, 2, 3}};                    // Center square
+
+  std::vector<std::array<size_t, 4>> corners{block0_corners, block1_corners,
+                                             block2_corners, block3_corners,
+                                             block4_corners};
+  // inner_radius is radius of the circle circumscribing inner square.
```

Comment:
This is documented in the options struct for inner radius, so no need for a comment